include/envoy/runtime/runtime.h

```diff
@@ -41,32 +42,33 @@ class Snapshot {
 public:
   virtual ~Snapshot() {}
 
-  /**
-   * The raw data from a single snapshot key.
-   */
   struct Entry {
-    /**
-     * The raw runtime data.
-     */
-    std::string string_value_;
-
-    /**
-     * The possibly parsed integer value from the runtime data.
-     */
+    enum class EntryType {
+      STRING_VALUE,
+      UINT_VALUE,
+      FRACTIONAL_PERCENT_VALUE,
+    };
+    std::string raw_string_value_;
     absl::optional<uint64_t> uint_value_;
+    absl::optional<envoy::type::FractionalPercent> fractional_percent_value_;
+    EntryType entry_type_;
```

Comment:
I'm not sure that it's worth maintaining this enum. All runtime values are effectively strings, and if they happen to parse to something else the user can check it via the optional value? WDYT?

Reply:
I just thought this way looked cleaner and added it to avoid the optional value check. In the event that more parsed types are added here, checking for optionals would quickly devolve into a mess.

Reply:
IMO I would remove for now, since it's possible also that in the future a value could be parsed as multiple types. I just don't think maintaining the secondary state is worth it. We can always add it back later?

Reply:
:+1: